public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -160,6 +179,48 @@ export const GeneratePopoverBody: React.FC<{
     return generateInsight();
   };
 
+  const handleNavigateToDiscover = async () => {
+    const context = await incontextInsight?.contextProvider?.();
+    const dsl = context?.additionalInfo?.dsl;
+    const indexName = context?.additionalInfo?.index;
+    if (!dsl || !indexName) return;
+    const dslObject = JSON.parse(dsl);
+    const filters = dslObject?.query?.bool?.filter;
+    if (!filters) return;
+    const timeDslIndex = filters?.findIndex((filter: Record<string, string>) => filter?.range);
+    const timeDsl = filters[timeDslIndex]?.range;
+    const timeFieldName = Object.keys(timeDsl)[0];
+    if (!timeFieldName) return;
+    filters?.splice(timeDslIndex, 1);
+
+    if (getStartServices) {
+      const [coreStart, startDeps] = await getStartServices();
+      const newDiscoverEnabled = coreStart.uiSettings.get(UI_SETTINGS.QUERY_ENHANCEMENTS_ENABLED);
+      if (!newDiscoverEnabled) {
+        // Only new discover supports DQL with filters.
+        coreStart.uiSettings.set(UI_SETTINGS.QUERY_ENHANCEMENTS_ENABLED, true);
+      }
```

Comment:
i think we can't turn on this setting on behalf of user. I would recommend we disable the navigate button and have a tooltip to let user to make the choice.

Reply:
I don't think we need to care this for now, for a POC, we already create index pattern for usar automatically. And new discover will be enabled by default in futre.

Reply:
> we already create index pattern for user automatically

I forget this.